include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -702,6 +702,23 @@ def TTNN_AllGatherOp: TTNN_Op<"all_gather"> {
     let hasVerifier = 1;
 }
 
+def TTNN_ScatterOp: TTNN_Op<"scatter"> {
+    let summary = "Scatter op.";
+    let description = [{
+      Embeds the values of the 'update' tensor into 'input' at the given index and puts the value in the 'output' tensor.
+      }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         AnyRankedTensor:$scatter_indices,
+                         AnyRankedTensor:$update,
+                         I32Attr:$dimension,
+                         AnyRankedTensor:$out);
```

Comment:
In TTNN lib, this op is implemented as a binary composite op. The op signature seems to be

```
    static Tensor invoke(
        const Tensor& input_tensor_a,
        const Tensor& input_tensor_b,
        const std::optional<MemoryConfig>& memory_config = std::nullopt) {
        return OpHandler<binary_comp_op_type>::handle(input_tensor_a, input_tensor_b, memory_config);
    }
```

Can you adapt this to match the signature above? We shouldn't be adding imaginary ops, the TTNN dialect should model TTNN library as closely as possible.

Reply:
Added scatter op as binary composite op, please review when you have the time.